### PR TITLE
heredoc: honor inline `COPY` irrespective of `.containerignore` file

### PR DESCRIPTION
**Pull request by flouthoc**
If we are copying heredoc files, we need to temporary place them in the context dir and then move to container via copier there are cases where .containerignore can have a patterns like which can match our heredoc files so let's not set any excludes or IgnoreFile for this copy.

Closes: https://github.com/containers/buildah/issues/5391

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note
heredoc: honor inline COPY irrespective of ignore files
```

### CODE REVIEW

**Comment by flouthoc** [comment]
@containers/buildah-maintainers PTAL

**Comment by rhatdan** [comment]
LGTM
@nalind @Luap99 @ashley-cui @giuseppe PTAL

File: imagebuildah/stage_executor.go [hunk 1]
```diff
@@ -543,6 +543,15 @@ func (s *StageExecutor) performCopy(excludes []string, copies ...imagebuilder.Co
 			StripSetuidBit:    stripSetuid,
 			StripSetgidBit:    stripSetgid,
 		}
+		if len(copy.Files) > 0 {
+			// If we are copying heredoc files, we need to temporary place
+			// them in the context dir and then move to container via copier
+			// there are cases where .containerignore can have a patterns like
+			// '*' which can match our heredoc files so let's not set any excludes
+			// or IgnoreFile for this copy.
+			options.Excludes = []string{}
```
Comment (nalind) [inline imagebuildah/stage_executor.go:552]: This can be `nil`, can't it?

Comment (flouthoc) [reply imagebuildah/stage_executor.go]: Yes `nil` sounds much better, done.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by flouthoc** [comment]
@containers/buildah-maintainers PTAL

**Pull request merged**